hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -58,26 +60,28 @@ class DefaultSource extends RelationProvider
       throw new HoodieException("'path' must be specified.")
     }
 
+    // Try to create hoodie table meta client from the give path
+    // TODO: Smarter path handling
+    val metaClient = try {
+      val conf = sqlContext.sparkContext.hadoopConfiguration
+      Option(new HoodieTableMetaClient(conf, path.get, true))
```

Comment:
would n't be problematic if `path` is a glob and not the actual basePath of the table? COW/Snapshot query can for e.g do this and I think we should handle the same for MOR as well.

Reply:
At this point we have:
- RO, Snapshot query for COW: Support glob and basePath
- Snapshot for MOR: only support basePath
- Incremental: Only support basePath

What I am trying to do here is:
- If the `path` contains glob, fall back to RO. This is the current behavior. Create metaClient will throw an Exception but handled below.
- If the `path` is basePath, we create the metaClient. If COW table, go RO relation. If MOR, go snapshot relation.

Reply:
>Snapshot for MOR: only support basePath

Let me think about this more. We need to support some form of globbing for MOR/Snapshot query.

Reply:
Udit's PR has this path handling. Should we merge part of his PR first? https://github.com/apache/hudi/pull/1702/files#diff-9a21766ebf794414f94b302bcb968f41R31
With this, we can handle user to `.load(basePath)` or `.load(basePath + "/*/*")` for COW, MOR and incremental.